### PR TITLE
HOTFIX update nupic build process

### PR DESCRIPTION
**Pull request by jcasner**

### CODE REVIEW

File: infrastructure/infrastructure/utilities/nupic/build_commands.py [hunk 1]
```diff
@@ -272,6 +272,8 @@ def buildNuPIC(env, logger):
       except OSError:
         # didn't exist, so just pass
         pass
+      # need to remove this folder to allow the caching process to work
+      shutil.rmtree("external/linux32arm")
```
Comment (oxtopus) [inline infrastructure/infrastructure/utilities/nupic/build_commands.py]: hrm, why?

Comment (jcasner) [inline infrastructure/infrastructure/utilities/nupic/build_commands.py]: There's something goofy with that folder and some invalid symlinks.  We used to have to remove it to even get the wheel to build, but Matt somehow resolved that.

Current issue:
```
build:INFO:2015-08-20 21:20:55,999:Caching NuPIC to /var/build/nupic/b6d6158424ac692689a5fa0b81f99aa5f9fce7b9
build:ERROR:2015-08-20 21:20:59,888:Caching NuPIC failed.
Traceback (most recent call last):
  File "/opt/numenta/products/infrastructure/infrastructure/utilities/nupic/build_commands.py", line 350, in cacheNuPIC
    shutil.copytree(env["NUPIC"], cachedPath)
  File "/opt/numenta/anaconda/lib/python2.7/shutil.py", line 208, in copytree
    raise Error, errors
Error: [('/opt/numenta/jenkins/workspace/nupic-product-pipeline/build-315/nupic/external/linux32arm/lib/python2.7', '/var/build/nupic/b6d6158424ac692689a5fa0b81f99aa5f9fce7b9/external/linux32arm/lib/python2.7', "[Errno 2] No such file or directory: '/opt/numenta/jenkins/workspace/nupic-product-pipeline/build-315/nupic/external/linux32arm/lib/python2.7'"), ('/opt/numenta/jenkins/workspace/nupic-product-pipeline/build-315/nupic/external/linux32arm/share/swig', '/var/build/nupic/b6d6158424ac692689a5fa0b81f99aa5f9fce7b9/external/linux32arm/share/swig', "[Errno 2] No such file or directory: '/opt/numenta/jenkins/workspace/nupic-product-pipeline/build-315/nupic/external/linux32arm/share/swig'")]
```
image
```
[ec2-user@ip-10-120-8-12 nupic]$ ls -al  /opt/numenta/jenkins/workspace/nupic-product-pipeline/build-315/nupic/external/linux32arm/lib/python2.7
lrwxrwxrwx. 1 ec2-user ec2-user 9 Aug 20 21:04 /opt/numenta/jenkins/workspace/nupic-product-pipeline/build-315/nupic/external/linux32arm/lib/python2.7 -> python2.6
```

Comment (oxtopus) [inline infrastructure/infrastructure/utilities/nupic/build_commands.py]: Why do we even have a linux32arm directory in this context?

Comment (jcasner) [inline infrastructure/infrastructure/utilities/nupic/build_commands.py]: No idea where that comes from.  @rhyolight can you help?

**Pull request merged**